reframe/frontend/statistics.py

```diff
@@ -84,6 +84,10 @@ def failure_report(self):
             report.append('  * System partition: %s' % partname)
             report.append('  * Environment: %s' % environ_name)
             report.append('  * Stage directory: %s' % check.stagedir)
+            if check.job:
+                report.append('  * Nodes: %s' % check.job._nodelist)
+            else:
+                report.append('  * Nodes: <%s>' % check.job)
```

Comment:
It be nicer if you can put everything in a single line:

```python
report.append('  * Node list: %s' % (','.join(check.job.nodelist) if check.job and check.job.nodelist else '<None>'))
```

Notice the `nodelist`  may also be `None`.